### ISSUE TITLE
MIDI interface only recongnizes one note at a time?

### ISSUE THREAD

**Issue by antoniograzioli** [VCVRack/Rack#228]
I am implementing a midi-trigger interface that sends simultaneous midi notes to different trig outs based on midi note numbers

It looks like the standard midi interface module, that I cloned to get the triggers, only accepts one note at a time (which is ok for a monosynth)...is it by design or what?

**Comment by AndrewBelt** [comment]
Yes, it's a monophonic MIDI Interface.

**Comment by antoniograzioli** [comment]
OK, so no way to make it poliphonic or send multiple trigger values...I'll see if I find a way to hack the code :-)

**Comment by AndrewBelt** [comment]
We're working on a Quad MIDI Interface, no need to duplicate work. #81

**Comment by antoniograzioli** [comment]
ok then

**Comment by rafzael** [comment]
wow it's a bit confusing, what do you mean by send midi notes to different trig outs? You ll need many cv outputs to have different notes, the trig will only send a trig signals. I believe you'll need to look at the code of the midi interface to see how it deals with midi input, and probably program a function that store the different midi notes message and note on/off message in an array with the same index, each time a new note is pressed you increase the index... each cv out will stay at the last value until a note on message has the same index.

**Comment by AndrewBelt** [comment]
Oh yes, you mean map about 16 MIDI note "on" messages to a trigger input. Yes, that's also planned but not explicitly written on that issue. Pinging @bontric in case he wants to work on that.

**Comment by antoniograzioli** [comment]
I don't need CV, only trigger to trigger drum samples.
thsi is my prototype, but if you send two simultaneous notes (which is standard in midi drumming) only one will play

see my prototype:
https://www.youtube.com/watch?v=-ozkmjwRc5U

**Comment by AndrewBelt** [comment]
Oh, that actually looks really nice, I'm impressed! If you'd like to collaborate with @bontric to get that merged in, I'll happily take it. I think our convention is to unfortunately not use LEDs beside the outputs singe plug lights make that (mostly) redundant, but I can be convinced either way. It's up to him really.

**Comment by antoniograzioli** [comment]
I'd be honored to contribute, but my coding skills are so limited, really...
I would be happy to send the code to @bontric  for  my basic MIDI-trigger interface (just promise he won't laugh when he sees my code patterns)
LEDs are mostly for aesthetics, and I wouldn't care less if you decice  to remove them in favour of "cables lights" or whatever

**Comment by bontric** [comment]
@antoniograzioli This sounds more like a problem with your implementation. Can you provide the code for your midi handling? Maybe I can help looking at it.

The rtMidi library used to handle midi is not limited to monophonic input, so I don't see a problem with that.

@AndreBelt I'll add a trigger module to the core soon, it's just a few lines of code, modifying the CC module

**Comment by antoniograzioli** [comment]
OK, but I tested the "standard" MIDI-CV interface in the sources and it does the same (only one note at a time is recongnized). ANd I bascially woorked on hacking the standard code for the MIDI-CV

Where can I send the code?

**Comment by AndrewBelt** [comment]
@bontric Okay, nice.

**Comment by bontric** [comment]
Yeah, but that's because the midi-to-cv module is supposed to be monophonic :)

Don't worry about your code quality, we all start somewhere and it's great that you actually try and provide some open source modules😉

**Comment by antoniograzioli** [comment]
ok here it is:

[Autodafe-MIDI.zip](https://github.com/VCVRack/Rack/files/1384179/Autodafe-MIDI.zip)

**Comment by bontric** [comment]
@antoniograzioli Yeah, so in this case you can not use the code from Midi-to-CV as it is more specific to a monophonic module. I added a trigger module to my pull request [here](https://github.com/VCVRack/Rack/pull/231). 
I hope this is what you where looking for and I also added velocity dependent output. 

In theory we could also add aftertouch for each note, but I don't have an aftertouch capable controller so I will add that later or if someone asks for it ;)
